Doc/library/typing.rst

```diff
@@ -1275,177 +1283,203 @@ These are not used in annotations. They are building blocks for creating generic
+   that the ``TypeVar`` will be solved using the most specific type possible::
 
-       def print_capitalized(x: S) -> S:
-           """Print x capitalized, and return x."""
-           print(x.capitalize())
-           return x
+      x = print_capitalized('a string')
+      reveal_type(x)  # revealed type is str
 
+      class StringSubclass(str):
+          pass
 
-       def concatenate(x: A, y: A) -> A:
-           """Add two strings or bytes objects together."""
-           return x + y
+      y = print_capitalized(StringSubclass('another string'))
+      reveal_type(y)  # revealed type is StringSubclass
 
-    Note that type variables can be *bound*, *constrained*, or neither, but
-    cannot be both bound *and* constrained.
+      z = print_capitalized(45)  # error: int is not a subtype of str
 
-    Bound type variables and constrained type variables have different
-    semantics in several important ways. Using a *bound* type variable means
-    that the ``TypeVar`` will be solved using the most specific type possible::
+   Type variables can be bound to concrete types, abstract types (ABCs or
+   protocols), and even unions of types::
 
-       x = print_capitalized('a string')
-       reveal_type(x)  # revealed type is str
+      U = TypeVar('U', bound=str|bytes)  # Can be any subtype of the union str|bytes
+      V = TypeVar('V', bound=SupportsAbs)  # Can be anything with an __abs__ method
 
-       class StringSubclass(str):
-           pass
+   .. _typing-constrained-typevar:
 
-       y = print_capitalized(StringSubclass('another string'))
-       reveal_type(y)  # revealed type is StringSubclass
+   Using a *constrained* type variable, however, means that the ``TypeVar``
+   can only ever be solved as being exactly one of the constraints given::
 
-       z = print_capitalized(45)  # error: int is not a subtype of str
+      a = concatenate('one', 'two')
+      reveal_type(a)  # revealed type is str
 
-    Type variables can be bound to concrete types, abstract types (ABCs or
-    protocols), and even unions of types::
+      b = concatenate(StringSubclass('one'), StringSubclass('two'))
+      reveal_type(b)  # revealed type is str, despite StringSubclass being passed in
 
-       U = TypeVar('U', bound=str|bytes)  # Can be any subtype of the union str|bytes
-       V = TypeVar('V', bound=SupportsAbs)  # Can be anything with an __abs__ method
+      c = concatenate('one', b'two')  # error: type variable 'A' can be either str or bytes in a function call, but not both
 
-.. _typing-constrained-typevar:
+   At runtime, ``isinstance(x, T)`` will raise :exc:`TypeError`.
 
-    Using a *constrained* type variable, however, means that the ``TypeVar``
-    can only ever be solved as being exactly one of the constraints given::
+   .. attribute:: __name__
+
+      The name of the type variable.
+
+   .. attribute:: __covariant__
+
+      Whether the type var has been explicitly marked as covariant.
```

Comment:
Since there's no such thing as implicit {co,contra,in}variance on 3.11:

```suggestion
      Whether the type var has been marked as covariant.
```